### PR TITLE
fix(carDealership): update due API schema changes

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

Update due car dealership [API schema changes](https://hedviginsurance.slack.com/archives/C0481RNNR8V/p1699521301552909).

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3535** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3535" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3535?utm_source=stack-comment).

File: apps/store/src/features/carDealership/ProductItemContractContainer.tsx [hunk 1]
```diff
@@ -12,20 +12,17 @@ export const ProductItemContractContainerCar = ({ contract }: Props) => {
...
   const { t } = useTranslation('carDealership')
 
-  // TODO: displayItems is failing in carTrialExtensionQuery
-  const productDetails = [
-    {
-      title: 'Registration number',
-      value: 'ABC 123',
-    },
-  ]
+  const productDetails = contract.displayItems.map((item) => ({
```
Comment (guilhermespopolin) [inline apps/store/src/features/carDealership/ProductItemContractContainer.tsx:15]: API is sucessfully returning this now so I took the opportunity to also include this.

**Review by robinandeer — APPROVED** [review]
Nice catch!

[goodcatch](https://app.graphite.dev/macros/HedvigInsurance)

**Review by alebedev — APPROVED** [review]
Nice, thank you!

**Pull request merged**